pkg/config/settings/runtine_setting_profiling_goroutines.go

```diff
@@ -0,0 +1,40 @@
+package settings
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// ProfilingGoroutines wraps runtime.SetBlockProfileRate setting
+type ProfilingGoroutines (string)
+
+// Name returns the name of the runtime setting
+func (r ProfilingGoroutines) Name() string {
+	return string(r)
+}
+
+// Description returns the runtime setting's description
+func (r ProfilingGoroutines) Description() string {
+	return "This setting controls whether internal profiling will collect goroutine stacktraces (requires profiling restart)"
+}
+
+// Hidden returns whether or not this setting is hidden from the list of runtime settings
+func (r ProfilingGoroutines) Hidden() bool {
+	return true
+}
+
+// Get returns the current value of the runtime setting
+func (r ProfilingGoroutines) Get() (interface{}, error) {
+	return config.Datadog.GetBool("internal_profiling.enable_goroutine_stacktraces"), nil
+}
+
+// Set changes the value of the runtime setting
+func (r ProfilingGoroutines) Set(value interface{}) error {
+	enabled, err := GetBool(value)
+	if err != nil {
+		return err
+	}
+
+	config.Datadog.Set("internal_profiling.enable_goroutine_stacktraces", enabled)
```

Comment:
Is `config.Datadog.Set` thread safe?

Reply:
[Yes](https://github.com/DataDog/datadog-agent/blob/971434bcef8ff208132d2b723ed63923bc907cd6/pkg/config/viper.go#L34)